pkg/operator/ceph/cluster/controller.go

```diff
@@ -338,24 +325,17 @@ func (r *ReconcileCephCluster) reconcileDelete(cephCluster *cephv1.CephCluster)
 }
 
 // NewClusterController create controller for watching cluster custom resources created
-func NewClusterController(context *clusterd.Context, rookImage string, volumeAttachment attachment.Attachment, operatorConfigCallbacks []func() error, addClusterCallbacks []func() error) *ClusterController {
+func NewClusterController(context *clusterd.Context, rookImage string, volumeAttachment attachment.Attachment) *ClusterController {
 	return &ClusterController{
-		context:                 context,
-		volumeAttachment:        volumeAttachment,
-		rookImage:               rookImage,
-		clusterMap:              make(map[string]*cluster),
-		operatorConfigCallbacks: operatorConfigCallbacks,
-		addClusterCallbacks:     addClusterCallbacks,
-		csiConfigMutex:          &sync.Mutex{},
+		context:          context,
+		volumeAttachment: volumeAttachment,
+		rookImage:        rookImage,
+		clusterMap:       make(map[string]*cluster),
+		csiConfigMutex:   &sync.Mutex{},
 	}
 }
 
 func (c *ClusterController) reconcileCephCluster(clusterObj *cephv1.CephCluster, ownerInfo *k8sutil.OwnerInfo) error {
-	if clusterObj.Spec.CleanupPolicy.HasDataDirCleanPolicy() {
```

Comment:
If the cleanup policy is set, we no longer want to skip the reconcile? Or how is this handled now?

Reply:
Yes, I'm catching this in the predicate instead here https://github.com/rook/rook/pull/8607/files#diff-9ca70f526fa8c8dd6b004cea5a40da0ba139b2d63fcf45998ed3296aa7cbb59fR136. I believe this is what should have been done originally.

Reply:
What if the operator is restarted? Do we need to keep this check here in that case?

Reply:
Good point, I added the same check in the CREATE event of the predicate

Reply:
Actually reverted. Thanks for pointing out